src/lib/jsonParseV2/parseMetadataCompoundValue.cpp

```diff
@@ -0,0 +1,337 @@
+
+      cvnP->name       = iter->name.GetString();
+      cvnP->container  = parent;
+      cvnP->rootP      = parent->rootP;
+      cvnP->level      = parent->level + 1;
+      cvnP->siblingNo  = counter;
+      cvnP->path       = parent->path + cvnP->name;
+
+      if (nodeType == "String")
+      {
+        cvnP->stringValue       = iter->value.GetString();
+      }
+      else if (nodeType == "Number")
+      {
+        cvnP->numberValue = iter->value.GetDouble();
+      }
+      else if ((nodeType == "True") || (nodeType == "False"))
+      {
+        cvnP->boolValue   = (nodeType == "True")? true : false;
+      }
+      else if (nodeType == "Null")
+      {
+        cvnP->valueType = orion::ValueTypeNone;
+      }
+      else if (nodeType == "Object")
+      {
+        cvnP->path += "/";
+        cvnP->valueType = orion::ValueTypeObject;
+      }
+      else if (nodeType == "Array")
+      {
+        cvnP->path += "/";
+        cvnP->valueType = orion::ValueTypeVector;
+      }
+
+      parent->childV.push_back(cvnP);
+        
+      //
+      // Recursive call if Object or Array
+      //
+      if ((nodeType == "Object") || (nodeType == "Array"))
+      {
+        parseMetadataCompoundValue(iter, mdP, cvnP);
+      }
+
+      ++counter;
+    }
+  }
+  else if (node->IsArray())
+  {
+    int counter  = 0;
+
+    for (Value::ConstValueIterator iter = node->Begin(); iter != node->End(); ++iter)
+    {
+      std::string                nodeType  = jsonParseTypeNames[iter->GetType()];
+      orion::CompoundValueNode*  cvnP      = new orion::CompoundValueNode();
+      char                       itemNo[4];
+
+      snprintf(itemNo, sizeof(itemNo), "%03d", counter);
+
+      cvnP->valueType  = stringToCompoundType(nodeType);
+      cvnP->container  = parent;
+      cvnP->rootP      = parent->rootP;
+      cvnP->level      = parent->level + 1;
+      cvnP->siblingNo  = counter;
+      cvnP->path       = parent->path + "[" + itemNo + "]";
+
+      if (nodeType == "String")
+      {
+        cvnP->stringValue       = iter->GetString();
+      }
+      else if (nodeType == "Number")
+      {
+        cvnP->numberValue = iter->GetDouble();
+      }
+      else if ((nodeType == "True") || (nodeType == "False"))
+      {
+        cvnP->boolValue   = (nodeType == "True")? true : false;
+      }
+      else if (nodeType == "Null")
+      {
+        cvnP->valueType = orion::ValueTypeNone;
+      }
+      else if (nodeType == "Object")
+      {
+        cvnP->path += "/";
+        cvnP->valueType = orion::ValueTypeObject;
+      }
+      else if (nodeType == "Array")
+      {
+        cvnP->path += "/";
+        cvnP->valueType = orion::ValueTypeVector;
+      }
+
+      parent->childV.push_back(cvnP);
+
+      //
+      // Recursive call if Object or Array
+      //
+      if ((nodeType == "Object") || (nodeType == "Array"))
+      {
+        parseMetadataCompoundValue(iter, mdP, cvnP);
+      }
+
+      ++counter;
+    }
+  }
+
```

Comment:
We don't really have a strict rule for this.
First of all, in this exact case, there is no reason to have two blank lines, and I will fix it.

I use two blank lines to separate 'logical groups' of code inside a function. A 'code-group' always starting with a comment, of course. Not sure this is mentioned in the style guide